Modix.Data/Models/Core/GuildChannelEntity.cs

```diff
@@ -1,48 +1,36 @@
-﻿using System.ComponentModel.DataAnnotations;
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 using System.ComponentModel.DataAnnotations.Schema;
 
 using Microsoft.EntityFrameworkCore;
-
-using Modix.Data.Utilities;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
 
 namespace Modix.Data.Models.Core
 {
-    /// <summary>
-    /// Describes information about a message channel, within a Discord guild managed by the application,
-    /// which may or may not still exist.
-    /// </summary>
     public class GuildChannelEntity
     {
-        /// <summary>
-        /// The snowflake ID of the channel, within the Discord API.
-        /// </summary>
         [Key]
-        [Required]
         [DatabaseGenerated(DatabaseGeneratedOption.None)]
         public ulong ChannelId { get; set; }
 
-        /// <summary>
-        /// The snowflake ID of the guild to which this channel belongs.
-        /// </summary>
-        [Required]
         public ulong GuildId { get; set; }
 
-        /// <summary>
-        /// The last-known name of the channel.
-        /// </summary>
         [Required]
```

Comment:
If we're removing the other `[Required]` tags, should we remove this one as well? EF recognizes nullable-reference-type annotations, right? That would make this equally redundant as the rest.

Reply:
I've only removed them for the non null types where they're required by default. I was going to bring this up as a discussion point. Being explicit versus not.

Reply:
My choice would either be to leave them all, or remove them all unless explicitly necessary, probably leaning toward the latter.

A quick test shows that they're definitely not necessary with nullable-reference-types enabled.

Food for thought, not really relevant to this PR.